### DISCUSSION TITLE
pandasq or dataframe_sql or duckdb l for Python 3.12 AWS Layer

### DISCUSSION THREAD

**Discussion by LyleMartin** [keithrozario/Klayers#417] [Q&A] [answered]
I would be eternally grateful if someone could add pandasql to this repository for python 3.12 (not arm). 

I have been trying for months to get a set of layers that work for my Lambda function with no luck. I cannot get the right environment to create them, or create them correctly. Always get errors.  

Please and thank you.

**Comment by LyleMartin** [comment]
Or dataframe_sql or duckdb. I have rewritten my code to work with these, trying different library combos. I have used AWS's library for pandas and the libraries here for pandas and numpy. But when I add pandasql or dataframe_sql or duckdb as another layer I get errors. Tried different dir structures with /lib/python3.12/site-packages -- and without. Duckdb showed real promise but I get the error "Unable to import module 'lambda_function': No module named 'duckdb.duckdb'"

Any help or direction would be greatly appreciated. I worked on this a year ago and gave up, then again 6 months ago, couple of months ago. I am not a Python expert, my code is fairly simple, just using pandas, numpy and one of  (pandasql, dataframe_sql, duckdb). My current running code in version Python 3.8 uses pandasql and I have faith in that. But trying to move to newer version there are all kinds of compatability issues. Have gotten circular import errors. But most recent is the error I mention above. 

Please and thank you for your help.

**Comment by keithrozario** [comment, marked as answer]
You can try making a pull request for those packages here, and the pipeline will build it once I merge. 

That might help solve the issue.

**Reply by LyleMartin** [reply]
Thank you
